src/main/java/org/ballerinalang/command/util/ToolUtil.java

```diff
@@ -477,31 +470,106 @@ private static void downloadAndSetupDist(PrintStream printStream, HttpURLConnect
-                    + File.separator + "launcher" + File.separator + OSUtils.getLangServerLauncherName());
-            File debugAdpater = new File(langServerPath + File.separator + "debug-adapter"
-                    + File.separator + "launcher" + File.separator + OSUtils.getDebugAdapterName());
+        } catch (NoSuchAlgorithmException e) {
+            throw ErrorUtil.createCommandException("unable to generate hashes for the distribution: " + e.getMessage());
+        } catch (IOException e) {
+            throw ErrorUtil.createCommandException("Ballerina distribution not found: " + e.getMessage());
+        } finally {
+            conn.disconnect();
+        }
+    }
 
-            if (debugAdpater.exists()) {
-                addExecutablePermissionToFile(debugAdpater);
-            }
+    private static String getDistSHA(String distribution) {
+        HttpURLConnection conn = null;
+        String hash = "";
+        try {
+            URL url = new URL(getServerURL() + "/distributions");
+            conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
+            conn.setRequestProperty("user-agent",
+                    OSUtils.getUserAgent(getCurrentBallerinaVersion(),
+                            getCurrentToolsVersion(), "jballerina"));
+            conn.setRequestProperty("Accept", "application/json");
+            if (conn.getResponseCode() != 200) {
+                conn.disconnect();
+                throw ErrorUtil.createCommandException(getServerRequestFailedErrorMessage(conn));
+            } else {
+                String json = convertStreamToString(conn.getInputStream());
+                Matcher matcher = Pattern.compile("\"version\":\"(.*?)\"").matcher(json);
+                int i = 0;
+                while (matcher.find()) {
+                    if (!matcher.group(1).equals(distribution)) {
+                        i++;
+                    } else {
+                        break;
+                    }
+                }
 
-            if (launcherServer.exists()) {
-                addExecutablePermissionToFile(launcherServer);
+                matcher = Pattern.compile("\"hash\":\"(.*?)\"").matcher(json);
+                int j = 0;
+                while (matcher.find()) {
+                    if (j==i) {
+                        hash = matcher.group(1);
+                        break;
+                    } else {
+                        j++;
+                    }
+                }
             }
-
-            new File(zipFileLocation).delete();
+        } catch (IOException e) {
+            throw ErrorUtil.createCommandException(CONNECTION_ERROR_MESSAGE);
         } finally {
-            conn.disconnect();
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+        return hash;
+    }
+
+    private static String calcDistSHA(String zipFileLocation) throws NoSuchAlgorithmException, IOException {
```

Comment:
shall we use generateDistributionHash as the name? The algorithm (SHA, MD) is defined by the implementation